frontend/src/metabase/dashboard/hooks/use-location-sync.ts

```diff
@@ -0,0 +1,76 @@
+import type { Location } from "history";
+import { useEffect, useMemo } from "react";
+import { replace } from "react-router-redux";
+import { usePrevious } from "react-use";
+import { omit } from "underscore";
+
+import { DEFAULT_EMBED_DISPLAY_OPTIONS } from "metabase/dashboard/hooks/use-embed-display-options";
+import { parseHashOptions, stringifyHashOptions } from "metabase/lib/browser";
+import { useDispatch } from "metabase/lib/redux";
+import { isNullOrUndefined } from "metabase/lib/types";
+
+const DEFAULT_DASHBOARD_EMBED_DISPLAY_OPTIONS: Record<string, any> = {
```

Comment:
Could we use `EmbedDisplayParams & {fullscreen: boolean, refresh: number | null}` here?

Reply:
Yeah that's better. Not sure what was going through my mind there lol